numerical_methods/jacobi_method.cpp

```diff
@@ -0,0 +1,158 @@
+/**
+ * @file
+ * @brief Implementation of The Jacobi Method
+ * https://en.wikipedia.org/wiki/Jacobi_method#Algorithm
+ * @details
```

Comment:
You might want to add more details about the algorithm. The mathematical details seem to be already there; great work!